cosmic-core/utils/src/main/java/com/cloud/utils/net/NetUtils.java

```diff
@@ -1504,4 +1574,25 @@ public static long getCidrSize(final String netmask) {
     public static enum SupersetOrSubset {
         isSuperset, isSubset, neitherSubetNorSuperset, sameSubnet, errorInCidrFormat
     }
+
+    public static Boolean validIpRangeList( final String ipRangeList ) {
+        if(StringUtils.isEmpty(ipRangeList)) {
+            return false;
+        }
+
+        String[] ipRanges = ipRangeList.split(",");
+
+        for (String ipRange : ipRanges) {
+            if(ipRange.contains("-")) {
```

Comment:
Lines 1586 to 1594 can be simplified to `if ((ipRange.contains("-") && !validIpRange(ipRange)) || !isValidIp(ipRange)) { return false }`